lib/github/services/repository.ts

```diff
@@ -1,39 +1,128 @@
 import { db } from "@/lib/db";
 
-export const selectRepository = async (id: string) => {
+// Activate a single repository with updated checks
+export const activateRepository = async (id: string, userId: string) => {
   try {
-    const selectedRepository = await db.repository.update({
-      where: {
-        id,
+    const repository = await db.repository.findUnique({
+      where: { id },
+      include: {
+        installation: {
+          include: {
+            memberships: true,
+          },
+        },
       },
-      data: {
-        configured: true,
+    });
+
+    if (!repository) {
+      throw new Error("Repository not found.");
+    }
+
+    const { installation } = repository;
+    let userHasPermission = false;
+
+    // Check if the installation is of type 'organization'
+    if (installation.type === "organization") {
```

Comment:
in other services we handle authorization in a separate util function (e.g. `auth.ts`) and keep the service without authentication. The advantage of this is that if needed an action could also be performed without a user (e.g. in a cron job or other system action).
Both is ok, but it should be consistent.

Reply:
Also it would be more secure to derive the userId from the session since this makes sure that it's not possible to somehow tinker with the call and add a different userId from your own. Here on the server it's not an issue but it might be hard to enforce that this doesn't open up a security issue on the frontend side (which it does in this case)